### PR TITLE
Patis branch to learn how to be polite

### PR DESCRIPTION
**Pull request by patiem**

### CODE REVIEW

File: client/src/main/java/connection/chain/ChainConfigFactory.java [hunk 1]
```diff
@@ -2,10 +2,9 @@
 
 public class ChainConfigFactory {
 
-  private ChainConfigFactory() {
-  }
+private ChainConfigFactory() {}
 
-  public static Chain configureChainOfResponsibilities() {
+public static Chain configureChainOfResponsibilities() {
```
Comment (M-Aarsman) [inline client/src/main/java/connection/chain/ChainConfigFactory.java]: javadoc

File: client/src/main/java/model/Ship.java [hunk 2]
```diff
@@ -9,15 +9,14 @@
 /**
  * It holds information on a particular ship.
  *
- * @author Patrycja Mikulska
  * @version 1.5
  */
-class Ship implements Iterable<Field>{
+public class Ship implements Iterable<Field>{
```
Comment (M-Aarsman) [inline client/src/main/java/model/Ship.java]: why you add public?

File: client/src/main/java/model/Ship.java [hunk 3]
```diff
@@ -9,15 +9,14 @@
...
 
   private final List<Field> masts;
   private final int shipLength;
 
-  Ship(int length) {
+  public Ship(int length) {
```
Comment (M-Aarsman) [inline client/src/main/java/model/Ship.java]: javadoc

File: client/src/main/java/model/ShipBoardUpdater.java [hunk 4]
```diff
@@ -4,38 +4,56 @@
...
+ *
+ * @version 1.5
+ */
+public class ShipBoardUpdater {
+  private PossiblePositions possiblePositions;
+  private ShipBoundariesPositions boundariesPositions;
+  private SeaCleaner seaCleaner;
+  private Ship lastShip;
+
+  public ShipBoardUpdater(PossiblePositions possiblePositions,
```
Comment (M-Aarsman) [inline client/src/main/java/model/ShipBoardUpdater.java]: javadoc

File: client/src/main/java/model/ShipBoundariesPositions.java [hunk 5]
```diff
@@ -8,15 +8,16 @@
...
- * @author Patrycja Mikulska
  * @version 1.5
  */
 public class ShipBoundariesPositions {
 
   private Set<Position> boundaries;
+  private Sea sea;
 
-  public ShipBoundariesPositions() {
+  public ShipBoundariesPositions(Sea sea) {
```
Comment (M-Aarsman) [inline client/src/main/java/model/ShipBoundariesPositions.java]: javadoc

File: client/src/test/java/model/DataProvider.java [hunk 6]
```diff
@@ -24,6 +23,33 @@
...
+  @org.testng.annotations.DataProvider(name = "fieldsOnBoundaries")
+  public static Object[] fieldsOnBoundaries() {
+
+    return  new Object[][]{ {0, 2}, { 0, 5}, {0, 7},
+              {9, 1}, { 9, 3}, {9, 6},
+              {2, 0}, { 4, 0}, {6, 0},
+              {1, 9}, { 3, 9}, {8, 9} };
+  }
+
+  @org.testng.annotations.DataProvider(name = "fieldsInCorners")
```
Comment (M-Aarsman) [inline client/src/test/java/model/DataProvider.java]: why @org.testng.annotations.DataProvider not just @DataProvider and import?

File: client/src/test/java/model/FleetTest.java [hunk 7]
```diff
@@ -46,4 +47,36 @@ public void returnsPositionsOfAllMast() {
     //then
     assertThat(returnedPositions).isEqualTo(expectedPositions);
   }
-}
\ No newline at end of file
+
+  @Test
+  public void descriptiveNameGoesHere() {
```
Comment (M-Aarsman) [inline client/src/test/java/model/FleetTest.java]: such a weird name for test :D

File: client/src/test/java/model/FleetTest.java [hunk 8]
```diff
@@ -46,4 +47,36 @@ public void returnsPositionsOfAllMast() {
...
+
+      //given
+    Position one = new Position(1);
```
Comment (M-Aarsman) [inline client/src/test/java/model/FleetTest.java]: maybe extract some method out of this?

File: client/src/test/java/model/PositionTest.java [hunk 9]
```diff
@@ -0,0 +1,85 @@
+package model;
+
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class PositionTest extends DataProvider{
+
```
Comment (M-Aarsman) [inline client/src/test/java/model/PositionTest.java]: why are methods names start from capital letter?

File: client/src/test/java/model/FleetTest.java (hunk 7 above)
Comment (patiem) [reply client/src/test/java/model/FleetTest.java]: this method should be removed

File: client/src/test/java/model/DataProvider.java (hunk 6 above)
Comment (patiem) [reply client/src/test/java/model/DataProvider.java]: Import doesnt work I dont know why

File: client/src/main/java/connection/Client.java [hunk 10]
```diff
@@ -16,7 +16,6 @@
 /**
  * It sends and receives messages from the server.
```
[the remark this answers is not in the document]
Comment (patiem) [reply client/src/main/java/connection/Client.java]: x

File: client/src/main/java/connection/chain/ChainConfigFactory.java (hunk 1 above)
Comment (patiem) [reply client/src/main/java/connection/chain/ChainConfigFactory.java]: Not my code, will be added by Bartek in his pull request

File: client/src/main/java/model/Ship.java (hunk 2 above)
Comment (patiem) [reply client/src/main/java/model/Ship.java]: I change place where Ships are created

File: client/src/test/java/model/PositionTest.java (hunk 9 above)
Comment (patiem) [reply client/src/test/java/model/PositionTest.java]: my mistake

File: client/src/test/java/model/FleetTest.java (hunk 8 above)
Comment (patiem) [reply client/src/test/java/model/FleetTest.java]: removed

**Pull request merged**